### ISSUE TITLE
[BUG] Can't be added to or create groups

### ISSUE THREAD

**Issue by nwlyoc** [nanu-c/axolotl#727]
I'm using Axolotl 1.0.9 on Manjaro and I still have (always had) the problem that I can't be added to groups. When people try to create a group with me as a member (in Signal iOS), they are told the group cannot be created because one of the members (me) uses an old version of Signal and has to upgrade first. I thought support for groupsv2 had been added for quite a while now? Also other bug reports here suggest that it works for them in one way or the other (even if there still may be problems like in #572). I tried un- and reregistering my number in Axolotl, but that didn't help. How is this supposed to work? Is there any way for me to debug this further (nothing much to see on my side since nothing regarding the group gets through to me)?

Trying to create a group myself doesn't work either. When I try to add members, none of my contacts are shown. Only contacts who don't have Signal show up (if I happen to have any in my Axolotl contact list), but obviously I can't add them.

**Comment by nanu-c** [comment]
I have look into this and the problem is, we are currently unable to set a flag for supporting groupsv2 on the signal server.

For a starting point we can look into this:
https://github.com/nanu-c/axolotl/blob/043afe184b5c9405270d754c2dd2bc09f8993b40/app/worker/textsecureapi.go#L234-L245

**Comment by nwlyoc** [comment]
I assume [this commit](https://github.com/signal-golang/textsecure/commit/0831947160af7c2f29c369b08fa9a16627d0631a) is meant to address this issue? Unfortunately the problem persists for me with Axolotl 1.1.0, other people still can't add me to groups and are told I would need to update Signal to a newer version.

**Comment by nwlyoc** [comment]
I installed v1.1.0 today and sent someone a message asking them to try to create a group with me. So I guess they synchronized my profile by then. That's when they encountered the known message about me needing to update Signal.
I did however notice that in my config.yaml gv2_2, gv2_3 and announcementgroup were set to false. Do they need to be set to true? I changed those to true and asked to be invited again, but that was just now, have to wait for the other person to get back to me. If it still doesn't work, I'll reach out on via Matrix.

**Comment by nwlyoc** [comment]
I finally managed to participate in group chats. I believe the problem was that the last time I used my number on the official Signal client was years ago when groupsv2 hadn't been introduced yet. I think Axolotl never correctly communicated my number as being groupsv2 ready. I recently installed Signal via Waydroid and registered my number there. I didn't even write or receive any messages there, but shortly after registered in on Axolotl again. Since then I can be added to groups by other people.
Around the same time I also upgraded from Axolotl 1.3.0 to 1.5.0, so maybe this was also responsible for the change, but the changelog doesn't suggest a change in that regard.

I still can't add members to a group when creating one (my Signal contacts don't show up), so I'm leaving this open.

**Comment by nuehm-arno** [comment]
Still an issue with v1.6.0?

**Comment by nwlyoc** [comment]
> Still an issue with v1.6.0?

Yes. It still says "Add contacts first" and none of my contacts show up.

**Comment by nanu-c** [comment]
To be added to groups is possible but group handling is completely missing. This has to be added again but will not be there until https://github.com/whisperfish/presage implements group handling support.
